### PR TITLE
Thumbnail - use Geoserver private urls (with admin)

### PR DESCRIPTION
**Pull request by DBlasby**
There seems to be an auth cache (geofence??) that is setting bad permissions for anonymous for 30 seconds. This uses the internal url (with admin auth) so it can see those layers.

There doesn't appear to be an issue with WMS GetMap.

### CODE REVIEW

File: mapstory/apps/thumbnails/tasks.py [hunk 1]
```diff
@@ -57,13 +58,37 @@ def run_process(self, args, env={}, timeout=66):
...
+                url,
+                {},
+                None,
+                None,
+                http_client
+            )
+        )
+        resp, content = http_client.request(url)
+        return content
+
```
Comment (cuttlefish) [inline mapstory/apps/thumbnails/tasks.py:83]: Can you drop the leading underscores from these? I think that only means something for object instance vars.

File: mapstory/apps/thumbnails/tasks.py [hunk 2]
```diff
@@ -229,7 +255,7 @@ def run(self, pk, overwrite=False):
             print "EXCEPTION - thumbnail generation"
             print(e)
             print traceback.format_exc()
-            raise
+            self.retry(max_retries=5, countdown=31) # retry in 31 seconds (auth cache timeout)
```
Comment (cuttlefish) [inline mapstory/apps/thumbnails/tasks.py:257]: Does the auth caching problem always take 30 seconds, or is that the worst case?
If it's the latter, we could probably try it every 5s or something and just have more retries. (unless hitting it early causes problems)

**Review by cuttlefish — CHANGES_REQUESTED** [review]

File: mapstory/apps/thumbnails/tasks.py (hunk 1 above)
Comment (DBlasby) [reply mapstory/apps/thumbnails/tasks.py:83]: I copied this from else where - changed.

File: mapstory/apps/thumbnails/tasks.py (hunk 2 above)
Comment (DBlasby) [reply mapstory/apps/thumbnails/tasks.py:257]: I appears to always take 30 seconds (although its no longer occurring).  The only errors that should be happening now are things like the not being able to get basemap images.  
I might be able to set this to 5 seconds with more retries and exponential delay.

**Review by cuttlefish — APPROVED** [review]

**Pull request merged**